apps/src/storage/dataBrowser/DataTableView.jsx

```diff
@@ -65,18 +67,23 @@ class DataTableView extends React.Component {
   };
 
   exportCsv = () => {
-    const isSharedTable =
-      this.props.tableListMap[this.props.tableName] === tableType.SHARED;
-    const tableName = encodeURIComponent(this.props.tableName);
-    const channelId = isSharedTable ? 'shared' : Applab.channelId;
-    location.href = `/v3/export-firebase-tables/${channelId}/${tableName}`;
+    // TODO: post-firebase-cleanup, remove this conditional, leave only logic from the else clause #56994
+    if (isFirebaseStorage()) {
+      const isSharedTable =
+        this.props.tableListMap[this.props.tableName] === tableType.SHARED;
+      const tableName = encodeURIComponent(this.props.tableName);
+      const channelId = isSharedTable ? 'shared' : Applab.channelId;
+      location.href = `/v3/export-firebase-tables/${channelId}/${tableName}`;
+    } else {
+      location.href = storageBackend().exportCsvUrl(this.props.tableName);
+    }
```

Comment:
Firebase CSV export redirects to the URL added by an old middleware method in tables_api.rb. In contrast, Datablock Storage redirects to a new method on DatablockStorageController.